src/types/composite/array.ts

```diff
@@ -173,7 +173,7 @@ export abstract class BasicArrayType<T extends ArrayLike<unknown>> extends Compo
     const remainder = size % 32;
     let i = 0;
     if (fullChunkCount > 0) {
-      for (const node of target.iterateNodesAtDepth(this.getChunkDepth(), 0, fullChunkCount)) {
+      for (const node of target.getNodesAtDepth(this.getChunkDepth(), 0, fullChunkCount)) {
```

Comment:
Can we benchmark using a regular `for (;;)` loop instead of `for of`? If we are removing iterators, we should go all the way.

Reply:
Ok, I'll remove `for of` in consumption of `getNodesAtDepth`. Further use of iterators should be evaluated separately since that will likely require slightly deeper code changes. Preference for keeping the PR light and not expanding the scope if possible.